lib/jsdom/living/nodes/HTMLInputElement-impl.js

```diff
@@ -408,6 +586,154 @@ class HTMLInputElementImpl extends HTMLElementImpl {
     }
   }
 
+  // https://html.spec.whatwg.org/multipage/input.html#dom-input-valueasdate
+  get valueAsDate() {
+    if (!allowValueAsDate(this.type)) {
+      return null;
+    }
+
+    const stringToDateConverter = this._convertStringToDate;
+    const window = this._ownerDocument._defaultView;
+    const convertedValue = stringToDateConverter(this._value);
+
+    if (convertedValue instanceof Date) {
+      return new window.Date(convertedValue.getTime());
+    }
+
+    return null;
+  }
+
+  set valueAsDate(v) {
+    if (!allowValueAsDate(this.type)) {
+      throw DOMException.create(this._globalObject, [
+        "Failed to set the 'valueAsDate' property on 'HTMLInputElement': This input element does not support Date " +
+        "values.",
+        "InvalidStateError"
+      ]);
+    }
+
+    if (v !== null && Object.prototype.toString.call(v) !== "[object Date]") {
```

Comment:
This&nbsp;should be&nbsp;using an&nbsp;`isDate(…)`&nbsp;helper, like&nbsp;what&nbsp;**WebIDL2JS** does&nbsp;with&nbsp;`isArrayBuffer(…)`: <https://github.com/kraynel/jsdom/pull/1>.